core/deployment/src/main/java/io/quarkus/deployment/CodeGenContext.java

```diff
@@ -26,30 +40,76 @@ public CodeGenContext(ApplicationModel model, Path outDir, Path workDir, Path in
         this.test = test;
     }
 
+    /**
+     * Application model
+     *
+     * @return application model
+     */
     public ApplicationModel applicationModel() {
         return model;
     }
 
+    /**
+     * Target directory for the generated output.
+     * The directory would typically be resolved as {@code <project.build.directory>/generated-sources/<codegen-provider-id>},
+     * where {@code <codegen-provider-id> would match the value of {@link CodeGenProvider#providerId()}.
+     * For example, for a code gen provider {@code foo}, the output directory in a typical Maven project would be
+     * {@code target/generated-sources/foo}.
+     *
+     * @return target directory for the generated output
+     */
     public Path outDir() {
         return outDir;
     }
 
+    /**
+     * Working directory, typically the main build directory of the project.
+     * For a typical Maven project it would be the {@code target} directory.
+     *
+     * @return working directory, typically the main build directory of the project
+     */
     public Path workDir() {
         return workDir;
     }
 
+    /**
+     * Directory containing input content for a code generator.
+     * For the main application build of a typical Maven project the input sources directory
+     * would be {@code <project.basedir>/src/main/<codegen-provider-id>}, while for the tests it would be
+     * {@code <project.basedir>/src/test/<codegen-provider-id>}, where {@code <codegen-provider-id}
+     * would match the value of {@link CodeGenProvider#providerId()}.
+     *
+     * @return directory containing input content a code generator
+     */
     public Path inputDir() {
         return inputDir;
     }
 
+    /**
+     * Whether the code generation process should inherit the launching process' output streams
+     * or redirect its output and error streams using {@link java.lang.ProcessBuilder.Redirect#PIPE}.
+     * In the current implementation this is typically set to {@code true} by the framework.
```

Comment:
Much better, thanks! 

If this is only important for CodeGenProviders launching new processeses, then we could perhaps s:

```suggestion
     * Whether any new processes spawned by the given CodeGenProvider should inherit the 
     * launching process' output streams
     * or redirect its output and error streams using {@link java.lang.ProcessBuilder.Redirect#PIPE}.
     * In the current implementation this is typically set to {@code true} by the framework.
```